rudra/utils/github_utils.py

```diff
@@ -0,0 +1,70 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Github utility providing functionality to read and write github resource."""
+from github import Github
+
+
+class GithubUtils:
+    """Class to perform action related to github like get content / raise PR etc."""
+
+    def __init__(self, github_token):
```

Comment:
It will be easy to use if the constructor takes upstream repo as well. It will avoid extra step of calling `set_repo`

Reply:
This was intentionally not done, so that same github object once initialized can be used with different flow in the code if required. Like say code needs to read two files from different repo like deployment yaml and repo config, than it will simple to create one object and set repo one by one to the same object and read required content.

Reply:
It is not a heavy weight object. I would still prefer to keep the lifetime of the object tided to a branch.